osmnx/features.py

```diff
@@ -680,7 +680,7 @@ def _filter_features(
         Filtered GeoDataFrame of features.
     """
     # remove any null or empty geometries then fix any invalid geometries
-    gdf = gdf[~(gdf["geometry"].isna() | gdf["geometry"].is_empty)]
```

Comment:
Isn't the `SettingWithCopyWarning` here a false positive? That is, we don't care about writing back to the original dataframe, because we're overwriting the variable itself. Given the expensiveness of `.copy()`, it may be better to filter/disable the warning on this line instead (assuming this is indeed a false positive).

Reply:
Also, regarding:

> Addressed a VisibleDeprecationWarning in the _verify_edge_attribute function by explicitly setting dtype=object for a NumPy array to handle ragged nested sequences, maintaining future compatibility.

How did you produce this warning? I'm using numpy 2 and haven't seen it. I'm trying to figure it out, but `dtype=object` doesn't make sense to me. This shouldn't be a ragged array, but rather always a $m \times 3$ array created from the tuples of `(u, v, attr_value)`.

Reply:
All warnings were from a CI run at that time on the main branch.

Unfortunately, at this time I don't have the capacity to follow up on this, so feel free to close or modify this PR.